src/model/ai/evaluation/BackRowCountEvaluator.java

```diff
@@ -22,18 +21,25 @@ public double evaluateBoard(Board theBoard, PieceColor color) {
         for (Square square : theBoard.getGameState()) {
```

Comment:
I love all of those ideas, but instead of refactoring BackRowCountEvaluator, I propose you just make another evaluator class with a slightly different and descriptive name, and then we can plug either of them in. Also, perhaps a whole different class just for intact back rows, i.e. IntactBackRowEvaluator.

In terms of not cycling through all of the squares, yes that would be more efficient, but we probably won't see any performance gain because only leaf nodes are ever evaluated in the search.

So I don't think I'll make any change to this class; agreed?